CMake/Version.cmake

```diff
@@ -0,0 +1,72 @@
+
+# Try to figure out  versioning scheme for dynamic libraries
+# based on the system we're compiling for
+function(determine_version_type _out_version_type_name)
+
+        set(_ver_type "none")
+
+        if(WIN32 OR CYGWIN)
+                set(_ver_type "windows")
+        else()
+                # this part tries to mimic autotools as close as possible
+                if(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Linux" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Haiku" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+                        set(_ver_type "linux")
+                elseif(CMAKE_SYSTEM_NAME STREQUAL "HP-UX" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "NetBSD" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+                        set(_ver_type "sunos")
+                elseif(CMAKE_SYSTEM_NAME STREQUAL "SCO_SV" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "UnixWare" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "UNIX_SV" OR
+                       CMAKE_SYSTEM_NAME STREQUAL "XENIX")
+                        set(_ver_type "sco")
+                elseif(APPLE)
+                        set(_ver_type "darwin")
+                elseif(QNXNTO)
+                        set(_ver_type "qnx")
+                endif()
+        endif()
+
+        set(${_out_version_type_name} "${_ver_type}" PARENT_SCOPE)
+endfunction()
+
+# Take the libtool-specific version info in form CURRENT:REVISION:AGE
+# and try to translate it to CMakes' VERSION / SOVERSION properties
+function(parse_versioninfo _versioninfo_string _out_version_name _out_soversion_name)
+        string(REPLACE ":" ";" _versioninfo_components ${_versioninfo_string})
+        list(LENGTH _versioninfo_components _len)
+        if(${_len} LESS 3)
+                message(FATAL_ERROR "Version string '${_versioninfo_string}' is not a valid libtool value.")
+        endif()
+
+        list(GET _versioninfo_components 0 _current)
+        list(GET _versioninfo_components 1 _revision)
+        list(GET _versioninfo_components 2 _age)
+
+        determine_version_type(_version_type)
+
+        if(_version_type STREQUAL "linux")
+                math(EXPR _major ${_current}-${_age})
+                set(${_out_version_name} ${_major}.${_age}.${_revision} PARENT_SCOPE)
+                set(${_out_soversion_name} ${_major} PARENT_SCOPE)
+        elseif(_version_type STREQUAL "sco" OR
+               _version_type STREQUAL "qnx")
```

Comment:
if "sco" and "qnx" behave the same, why aren't they then instead made the same type?

Reply:
Just because I've coded looking at autotools sources. I'm not sure if I need to distinguish them later. If not, they could be merged.